src/ContaoCommunityAlliance/DcGeneral/Contao/Dca/Builder/Legacy/LegacyDcaDataDefinitionBuilder.php

```diff
@@ -1618,4 +1622,47 @@ protected function evalFlag($config, $flag)
         $this->evalFlagGrouping($config, $flag);
         $this->evalFlagGroupingLength($config, $flag);
     }
+
+    /**
+     * Try to determine the empty type from SQL type.
+     *
+     * @param PropertyInterface $property The property to store the value into.
+     * @param string            $sqlType  The SQL type.
+     *
+     * @return void
+     */
+    private function determineEmptyValueFromSql(PropertyInterface $property, $sqlType)
+    {
+        if ($property instanceof EmptyValueAwarePropertyInterface) {
+            $property->setEmptyValue($this->getEmptyValueByFieldType($sqlType));
+        }
+    }
+
+    /**
+     * Return the empty value based on the SQL string.
+     *
+     * This method is backported from Contao 4.0 to have a value at hand.
+     *
+     * @param string $sql The SQL string.
+     *
+     * @return string|integer|null The empty value
+     */
+    private function getEmptyValueByFieldType($sql)
+    {
+        if (empty($sql)) {
+            return '';
+        }
+        if (stripos($sql, 'NOT NULL') === false) {
```

Comment:
Contao started to support array based sql syntax lately (not since when). I think we should support it as well.

Reply:
We will in C4, where this method will get dropped and the "real" Method from `\Widget` will get used.
We can not use that one in C 3.5 however, as this would introduce a dependency on doctrine DBAL then.